src/data.rs

```diff
@@ -154,6 +248,48 @@ mod array_impls {
         })))
     }
 
+    pub(crate) fn create_string_array_with_errors<DT: CellType + DataType + Debug>(
+        data: &Range<DT>,
+        col: usize,
+        offset: usize,
+        limit: usize,
+    ) -> (Arc<dyn Array>, CellErrors) {
+        let mut cell_errors = CellErrors { errors: vec![] };
+
+        let arr = Arc::new(StringArray::from_iter((offset..limit).map(|row| {
+            data.get((row, col)).and_then(|cell| {
+                if cell.is_empty() {
+                    None
+                } else if cell.is_string() {
+                    cell.get_string().map(str::to_string)
+                } else if cell.is_datetime() {
+                    cell.get_datetime()
+                        .and_then(|dt| dt.as_datetime())
+                        .map(|dt| dt.to_string())
+                } else if cell.is_datetime_iso() {
+                    cell.get_datetime_iso().map(str::to_string)
+                } else if cell.is_bool() {
+                    cell.get_bool().map(|v| v.to_string())
+                } else if cell.is_float() {
+                    cell.get_float().map(excel_float_to_string)
```

Comment:
It'd be great to have shared code between the regular and `_with_errors` implems here, could you please add a TODO comment so we take care of it at some point ?

Reply:
Added